controllers/clusterversion_controller.go

```diff
@@ -139,184 +141,225 @@ func (c *ClusterVersionReconciler) Reconcile(ctx context.Context, req ctrl.Reque
+		if err != nil {
+			c.log.Error(err, "unable to create rbd CSIDriver")
+			return ctrl.Result{}, err
+		}
+
+		prometheusRule := &monitoringv1.PrometheusRule{}
+		err = k8sYAML.NewYAMLOrJSONDecoder(bytes.NewBufferString(string(pvcPrometheusRules)), 1000).Decode(prometheusRule)
+		if err != nil {
+			c.log.Error(err, "Unable to retrieve prometheus rules.", "prometheusRule", klog.KRef(prometheusRule.Namespace, prometheusRule.Name))
+			return ctrl.Result{}, err
+		}
+
+		operatorConfig, err := c.getOperatorConfig()
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		prometheusRule.SetNamespace(c.OperatorNamespace)
+
+		err = c.createOrUpdate(prometheusRule, func() error {
+			applyLabels(operatorConfig.Data["OCS_METRICS_LABELS"], &prometheusRule.ObjectMeta)
+			return c.own(prometheusRule)
+		})
+		if err != nil {
+			c.log.Error(err, "failed to create/update prometheus rules")
+			return ctrl.Result{}, err
+		}
+
+		c.log.Info("prometheus rules deployed", "prometheusRule", klog.KRef(prometheusRule.Namespace, prometheusRule.Name))
+	} else {
+		// deletion phase
+		result, err := c.deletionPhase()
+		if err != nil || !result.Requeue || result.RequeueAfter > 0 {
+			return result, err
+		}
+
+		//remove finalizer
+		controllerutil.RemoveFinalizer(&instance, ClusterVersionFinalizer)
+		if err := c.Client.Update(c.ctx, &instance); err != nil {
+			return ctrl.Result{}, err
+		}
+		c.log.Info("finallizer removed successfully")
 	}
 
-	prometheusRule := &monitoringv1.PrometheusRule{}
-	err = k8sYAML.NewYAMLOrJSONDecoder(bytes.NewBufferString(string(pvcPrometheusRules)), 1000).Decode(prometheusRule)
-	if err != nil {
-		c.log.Error(err, "Unable to retrieve prometheus rules.", "prometheusRule", klog.KRef(prometheusRule.Namespace, prometheusRule.Name))
+	return ctrl.Result{}, nil
+}
+
+func (c *ClusterVersionReconciler) deletionPhase() (reconcile.Result, error) {
+	if err := csi.DeleteCSIDriver(c.ctx, c.Client, csi.GetCephFSDriverName()); err != nil && !k8serrors.IsNotFound(err) {
+		c.log.Error(err, "unable to delete cephfs CSIDriver")
 		return ctrl.Result{}, err
 	}
-
-	operatorConfig, err := c.getOperatorConfig()
-	if err != nil {
+	if err := csi.DeleteCSIDriver(c.ctx, c.Client, csi.GetRBDDriverName()); err != nil && !k8serrors.IsNotFound(err) {
+		c.log.Error(err, "unable to delete rbd CSIDriver")
 		return ctrl.Result{}, err
 	}
-	prometheusRule.SetNamespace(c.OperatorNamespace)
-
-	err = c.createOrUpdate(prometheusRule, func() error {
-		applyLabels(operatorConfig.Data["OCS_METRICS_LABELS"], &prometheusRule.ObjectMeta)
-		return c.own(prometheusRule)
-	})
-	if err != nil {
-		c.log.Error(err, "failed to create/update prometheus rules")
+	if err := c.Client.Delete(c.ctx, c.scc); err != nil && !k8serrors.IsNotFound(err) {
+		c.log.Error(err, "unable to delete SCC")
 		return ctrl.Result{}, err
 	}
-
-	c.log.Info("prometheus rules deployed", "prometheusRule", klog.KRef(prometheusRule.Namespace, prometheusRule.Name))
-
```

Comment:
https://github.com/red-hat-storage/ocs-client-operator/blob/a5c640e9dcf43bf9239bb75f436f553f316ae715/controllers/clusterversion_controller.go#L466-L472

The above doesnt have any ownerRef we might need to add to that as well?

Reply:
I think handling cleanup of console components should go into its own PR

Reply:
Yes we can , we just need to make sure we don't miss it out.

Reply:
I dont have an issue with adding an ownerRef, but since this is not related to this PR i'll open another one addressing it.